drivers/timer/arcv2_timer0.c

```diff
@@ -186,6 +219,33 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 	 * then shut off the counter. (Note: we can assume if idle==true
 	 * that interrupts are already disabled)
 	 */
+#ifdef CONFIG_SMP
+	if (IS_ENABLED(CONFIG_TICKLESS_IDLE) && idle && ticks == K_FOREVER) {
+		timer0_control_register_set(0);
+		timer0_count_register_set(0);
+		timer0_limit_register_set(0);
```

Comment:
Given ARC core timers are actually free running counters capable of interrupt generation on reaching a preset "limit" value, there's no point in messing with all 3 registers here. The only thing we need to do is clear `control.IE` bit so that IRQs are not generated, all the rest is waste of cycles.

Reply:
it's kind of optimization, but i prefer letting it in a determined status.

Reply:
Then add a comment explicitly saying what happens here like:

> In reality ARC core timers (Timer0 & Timer1) are always running and we cannot stop them. The only thing we may do is disable interrupt generation which is done by writing 0 to `IE` bit in timer's control register. But for XXX (nicer debugging experience whatever) we also set 0 in `count` & `limit` registers.

Still IMHO this code might fool a reader. Moreover on power-on timer has interrupts disabled `control.IE` = 0. That's what PRM says:
> When the processor is Reset, the IE flag is set to 0

So nothing needs to be done here at all.